client-misk/src/main/kotlin/app/cash/backfila/client/misk/client/BackfilaClientLoggingSetup.kt

```diff
@@ -0,0 +1,62 @@
+package app.cash.backfila.client.misk.client
+
+import misk.logging.getLogger
+import org.slf4j.MDC
+
+interface BackfilaClientLoggingSetupProvider {
+  fun <T>withBackfillRunLogging(backfillName: String, backfillId: String, wrapped: () -> T) : T
+  fun <T>withBackfillPartitionLogging(backfillName: String, backfillId: String, partitionName: String, wrapped: () -> T) : T
+}
+
+class BackfilaClientNoLoggingSetupProvider : BackfilaClientLoggingSetupProvider {
+  override fun <T>withBackfillRunLogging(backfillName: String, backfillId: String, wrapped: () -> T) : T {
+    return wrapped.invoke()
+  }
+
+  override fun <T>withBackfillPartitionLogging(backfillName: String, backfillId: String, partitionName: String, wrapped: () -> T) : T {
+    return wrapped.invoke()
+  }
+}
+
+class BackfilaClientMDCLoggingSetupProvider : BackfilaClientLoggingSetupProvider {
+  companion object {
```

Comment:
companion is at bottom usually

Reply:
I just copied an example from Misk but sure.